application/asset/js/resource-form.js

```diff
@@ -356,163 +365,189 @@
-            // selector.
-            fields.find('div.single-selector').hide();
-            fields.find('div.default-selector').show();
-            return;
-        }
-        var url = templateSelect.data('api-base-url') + '/' + templateId;
-        return $.get(url)
-            .done(function(data) {
-                if (changeClass) {
-                    // Change the resource class.
-                    var classSelect = $('#resource-class-select');
-                    if (data['o:resource_class'] && classSelect.val() === '') {
-                        classSelect.val(data['o:resource_class']['o:id']);
-                        classSelect.trigger('chosen:updated');
+
+        // Reset settings of the previous template.
+        $('#resource-values').data('template-settings', {});
+        fields.attr('data-template-id', '');
+        fields.data('settings', {});
+
+        // Fieldsets may have been marked as required or private in a previous state.
+        fields.removeClass('required');
+        fields.removeClass('private');
+
+        // Using the default resource template, so all properties should use the default
+        // selector.
+        fields.find('div.single-selector').hide();
+        fields.find('div.default-selector').show();
+
+        if (templateId) {
+            var url = templateSelect.data('api-base-url') + '/' + templateId;
+            $.get(url)
+                .done(function(data) {
+                    if (changeClass) {
+                        // Change the resource class.
+                        var classSelect = $('#resource-class-select');
+                        if (data['o:resource_class'] && classSelect.val() === '') {
+                            classSelect.val(data['o:resource_class']['o:id']);
+                            classSelect.trigger('chosen:updated');
+                        }
                     }
+
+                    // Store global settings of the template.
+                    $('#resource-values').data('template-settings', data['o:settings'] ? data['o:settings'] : {});
+
+                    // Rewrite every property field defined by the template. We
+                    // reverse the order so property fields on page that are not
+                    // defined by the template are ultimately appended.
+                    data['o:resource_template_property']
+                        .reverse().map(function(templateProperty) {
+                            rewritePropertyField(templateProperty);
+                        });
+                })
+                .fail(function() {
+                    console.log('Failed loading resource template from API');
+                })
+                .always(finalize);
+        } else {
+            finalize();
+        }
+
+        function finalize() {
+            // Remove empty properties, except the templates ones.
+            // TODO Keep properties selected by user (remove the "default-value"?).
```

Comment:
I don't think we want to have in the feature to remove empty properties if it doesn't handle keeping the user-selected ones.

Reply:
This is fixed below.